src/hotspot/cpu/arm/sharedRuntime_arm.cpp

```diff
@@ -819,21 +815,13 @@ nmethod* SharedRuntime::generate_native_wrapper(MacroAssembler* masm,
   // Unverified entry point
   address start = __ pc();
 
-  // Inline cache check, same as in C1_MacroAssembler::inline_cache_check()
   const Register receiver = R0; // see receiverOpr()
-  __ load_klass(Rtemp, receiver);
-  __ cmp(Rtemp, Ricklass);
-  Label verified;
+  __ verify_oop(receiver);
+  // Inline cache check
+  __ ic_check(CodeEntryAlignment /* end_alignment */);
 
-  __ b(verified, eq); // jump over alignment no-ops too
-  __ jump(SharedRuntime::get_ic_miss_stub(), relocInfo::runtime_call_type, Rtemp);
-  __ align(CodeEntryAlignment);
-
-  // Verified entry point
-  __ bind(verified);
   int vep_offset = __ pc() - start;
```

Comment:
Maybe add back
`// Verified entry point`
above this line.

Reply:
Sure.